src/main/java/org/alliancegenome/curation_api/services/AffectedGenomicModelService.java

```diff
@@ -73,47 +71,42 @@
 		return agmDAO.persist(agm);
 	}
 
+	@Override
 	@Transactional
-	public void removeOrDeprecateNonUpdated(Long id, String loadDescription) {
+	public ObjectResponse<AffectedGenomicModel> deleteById(Long id) {
+		deprecateOrDelete(id, true, "AGM DELETE API call", false);
+		ObjectResponse<AffectedGenomicModel> ret = new ObjectResponse<>();
+		return ret;
+	}
+
+	@Override
+	@Transactional
+	public AffectedGenomicModel deprecateOrDelete(Long id, Boolean throwApiError, String requestSource, Boolean forceDeprecate) {
 		AffectedGenomicModel agm = agmDAO.find(id);
 		if (agm != null) {
-			List<Long> referencingDAIds = agmDAO.findReferencingDiseaseAnnotations(id);
-			Boolean anyReferencingEntities = false;
-			for (Long daId : referencingDAIds) {
-				DiseaseAnnotation referencingDA = diseaseAnnotationService.deprecateOrDeleteAnnotationAndNotes(daId, false, loadDescription, true);
-				if (referencingDA != null) {
-					anyReferencingEntities = true;
-				}
-			}
-			List<Long> referencingPAIds = agmDAO.findReferencingPhenotypeAnnotations(id);
-			for (Long paId : referencingPAIds) {
-				PhenotypeAnnotation referencingPA = phenotypeAnnotationService.deprecateOrDeleteAnnotationAndNotes(paId, false, loadDescription, true);
-				if (referencingPA != null) {
-					anyReferencingEntities = true;
-				}
-			}
-			if (CollectionUtils.isNotEmpty(agm.getConstructGenomicEntityAssociations())) {
-				for (ConstructGenomicEntityAssociation association : agm.getConstructGenomicEntityAssociations()) {
-					association = constructGenomicEntityAssociationService.deprecateOrDeleteAssociation(association.getId(), false, loadDescription, true);
-					if (association != null) {
-						anyReferencingEntities = true;
-					}
-				}
-			}
-
-			if (anyReferencingEntities) {
+			if (forceDeprecate || agmDAO.hasReferencingDiseaseAnnotations(id) || agmDAO.hasReferencingPhenotypeAnnotations(id) ||
```

Comment:
🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.OperatorWrapCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'||' should be on a new line.